### ISSUE TITLE
Auto-generated SQL queries are incorrect

### ISSUE THREAD

**Issue by SananGuliyev** [TablePlus/TablePlus#554]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0): MySQL 5.7

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81): 115

3. The steps to reproduce this issue:

When I try to update row within the GUI it produces wrong `UPDATE` query even I only change one column the auto-generated query try to update multiple columns. 

![screen shot 2018-07-18 at 13 33 20](https://user-images.githubusercontent.com/5057582/42879046-83d6fdd0-8a8f-11e8-8c45-56eef0b5cc62.png)
![screen shot 2018-07-18 at 13 33 34](https://user-images.githubusercontent.com/5057582/42879047-83f78fa0-8a8f-11e8-86be-99b6c20a6698.png)

**Comment by huyphams** [comment]
Thanks @SananGuliyev TablePlus only generate the update statement for the modified field.
I believe there is a bug but with your limited info, I can't reproduce it (I tested with my setup and it generates correctly.)

**Comment by SananGuliyev** [comment]
Yes, you are right. I checked it and it generates update statement only for the modified fields but if there is no auto-increment id then it generates some `where` statement which avoids the update.

![screen shot 2018-07-18 at 13 40 34](https://user-images.githubusercontent.com/5057582/42879399-c350c332-8a90-11e8-914c-2f59b2bde85a.png)

**Comment by huyphams** [comment]
Because MySQL doesn't have any hidden row id (like ctid in PostgreSQL) so it needs a where condition to avoid multiple updates. (Without where, it updates all the rows).

**Comment by SananGuliyev** [comment]
But in the end, the affected rows is 0. I tried the same thing on `Sequel Pro` and it updated row as expected.

**Comment by huyphams** [comment]
Hey, could you go to Sequel Pro and copy the generated statement (click on console button on the top right), then compare with the statement which was generated by TablePlus. I believe we can figure out the bug easily. I can do it my self but probably your database contains sensitive information.
I appreciate your help

**Comment by SananGuliyev** [comment]
The difference is like that.

That's generated by `Sequel Pro`:
```UPDATE `table_name` SET `modified_column` = 'new_value' WHERE `unique_column` = 'unique_column_value';```

That's created by `TablePlus`:
```UPDATE `table_name` SET `modified_column` = 'new_value' WHERE `unique_column` = 'unique_column_value' AND `some_non_unique_column_1` = 'some_non_unique_column_1' AND `some_non_unique_column_2` = 'some_non_unique_column_2_value' ... (all columns with current data);```

**Comment by huyphams** [comment]
I figured out the bug, it because TablePlus tried to compare float value (we should not compare float value because it will be random false).
For example:

```
CREATE TABLE `test` (
  `the_float` float DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8

INSERT INTO `test` (`the_float`) VALUES ('1.1');
```
Now try to edit the value, it always fail, both Sequel Pro and TablePlus. As I said, optimize based on a unique index is not a full solution.
